### PR TITLE
DL-3660 Reduced warning count

### PR DESCRIPTION
**Pull request by jmountain-hmrc**
# DL-3660 Reduced warning count

**Code cleanup**

* Reduced compiliations warnings on awrs-frontend.

## Checklist

 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions
 - [x]  I've added my code using logical, atomic commits, squashing as appropriate - including the JIRA issue number in the commit message
 - [x]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/controllers/AddressLookupController.scala [hunk 1]
```diff
@@ -35,9 +35,9 @@ import scala.concurrent.{ExecutionContext, Future}
 
 class AddressLookupController @Inject()(mcc: MessagesControllerComponents,
                                         val authConnector: DefaultAuthConnector,
-                                        val addressLookupService: AddressLookupService,
+                                        val addressLookupService: AddressLookupConnector,
```
Comment (paulanderson1234) [inline app/controllers/AddressLookupController.scala:38]: Are we OK leaving reference to a service (in name only) where we have changed to a connector?
Not sure how much work is involve in changing this though

File: app/connectors/AddressLookupConnector.scala [hunk 2]
```diff
@@ -54,6 +51,6 @@ class AddressLookupService @Inject()(servicesConfig: ServicesConfig,
   }
 }
 
-trait HasAddressLookupService {
-  val addressLookupService: AddressLookupService
+trait HasAddressLookupConnector {
+  val addressLookupService: AddressLookupConnector
```
Comment (paulanderson1234) [inline app/connectors/AddressLookupConnector.scala:55]: Are we OK leaving reference to a service (in name only) where we have changed to a connector?
What we be involved in refactoring to val addressLookupConnector

File: app/controllers/AddressLookupController.scala (hunk 1 above)
Comment (jmountain-hmrc) [reply app/controllers/AddressLookupController.scala:38]: Ah, that can be fixed

File: app/controllers/ApplicationController.scala [hunk 3]
```diff
@@ -20,29 +20,25 @@ import config.ApplicationConfig
...
-    implicit request =>
-      Redirect(applicationConfig.signOut)
+  def timedOut(): Action[AnyContent] = Action { _ =>
+    Redirect(applicationConfig.signOut)
   }
 
-  def keepAlive: Action[AnyContent] = Action {
-    implicit request =>
-      Ok("OK")
+  def keepAlive: Action[AnyContent] = Action {_ =>
```
Comment (paulanderson1234) [inline app/controllers/ApplicationController.scala:40]: Did you want to space this consistently as per defs above?

File: app/controllers/ReapplicationController.scala [hunk 4]
```diff
@@ -30,7 +30,9 @@ import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
 import services.{DeEnrolService, KeyStoreService, Save4LaterService}
 import uk.gov.hmrc.play.bootstrap.auth.DefaultAuthConnector
 import uk.gov.hmrc.play.bootstrap.controller.FrontendController
-import utils.{AccountUtils, AwrsSessionKeys, LoggingUtils}
+import utils.AwrsSessionKeys._
+import utils.{AccountUtils, LoggingUtils}
```
Comment (paulanderson1234) [inline app/controllers/ReapplicationController.scala:34]: Do these need imports here if they are being injected implicitly?
Just observing that we don't appear to have done this in the many changes above

Comment (paulanderson1234) [reply app/controllers/ReapplicationController.scala:34]: Think I answered my own question ... tried removing and get not found so scratch this comment

File: app/controllers/WithdrawalController.scala [hunk 5]
```diff
@@ -47,7 +47,10 @@ class WithdrawalController @Inject()(mcc: MessagesControllerComponents,
                                      val authConnector: DefaultAuthConnector,
                                      val auditable: Auditable,
                                      val accountUtils: AccountUtils,
-                                     implicit val applicationConfig: ApplicationConfig) extends FrontendController(mcc) with AwrsController with LoggingUtils {
+                                     implicit val applicationConfig: ApplicationConfig,
+                                     templateWithdrawlReasons: views.html.awrs_withdrawal_reasons,
+                                     templateWithdrawlConfirmation: views.html.awrs_withdrawal_confirmation,
```
Comment (paulanderson1234) [inline app/controllers/WithdrawalController.scala:52]: Typo?
Did you mean templateWithdrawalConfirmation

File: app/models/StatusNotificationModel.scala [hunk 6]
```diff
@@ -40,18 +38,13 @@ object StatusContactType {
...
+    case JsString(code) => apply(code)
+    case _ => throw new RuntimeException(s"Unexpected StatusContactType: $json")
+  })
 
-  implicit val writer: Writes[StatusContactType] = new Writes[StatusContactType] {
-    def writes(v: StatusContactType): JsValue = v match {
-      case value => JsString(v.code)
-    }
+  implicit val writer: Writes[StatusContactType] = {
+    case v@_ => JsString(v.code)
```
Comment (paulanderson1234) [inline app/models/StatusNotificationModel.scala:47]: Just checking this was intended : v@_ ?

File: test/views/BusinessTypeViewTest.scala [hunk 7]
```diff
@@ -29,11 +30,11 @@ import services.{CheckEtmpService, ServicesUnitTestFixture}
 import uk.gov.hmrc.auth.core.{AffinityGroup, Enrolment, EnrolmentIdentifier, Enrolments}
 import utils.TestUtil._
 import utils.{AwrsUnitTestTraits, TestUtil}
-import uk.gov.hmrc.auth.core.retrieve.{GGCredId, ~}
+import uk.gov.hmrc.auth.core.retrieve.{Credentials, GGCredId, ~}
 
 import scala.concurrent.Future
 
-class BusinessTypeViewTest extends AwrsUnitTestTraits
+class BusinessTypeViewTest @Inject()(template: views.html.awrs_business_type) extends AwrsUnitTestTraits
```
Comment (paulanderson1234) [inline test/views/BusinessTypeViewTest.scala:37]: As discussed I think you wanted to use val template = injectInstanceOf instead here?

File: test/utils/AwrsUnitTestTraits.scala [hunk 8]
```diff
@@ -20,13 +20,15 @@ import audit.Auditable
 import config.ApplicationConfig
 import org.jsoup.nodes.Document
 import org.mockito.ArgumentMatchers
+import org.mockito.ArgumentMatchers.any
```
Comment (paulanderson1234) [inline test/utils/AwrsUnitTestTraits.scala:23]: would import org.mockito.ArgumentMatchers._ work here to replace the 2 imports

File: test/services/mocks/MockTestStatusManagementService.scala [hunk 9]
```diff
@@ -28,7 +28,7 @@ trait MockTestStatusManagementService extends AwrsUnitTestTraits
   with MockSave4LaterService
```
Comment (paulanderson1234) [inline test/services/mocks/MockTestStatusManagementService.scala:28]: If we have removed testSave4LaterService do we need it in the trait MockTestStatusManagementService ?
I commented it out (line28) and seems to have no effect - all unit and it tests run OK

File: test/services/EnrolServiceTest.scala [hunk 10]
```diff
@@ -21,7 +21,7 @@ import connectors.TaxEnrolmentsConnector
 import org.mockito.{ArgumentMatchers, Matchers}
```
Comment (paulanderson1234) [inline test/services/EnrolServiceTest.scala:21]: Matchers showing as unused import

File: test/services/CheckEtmpServiceTest.scala [hunk 11]
```diff
@@ -33,7 +33,7 @@ import utils.{AccountUtils, TestUtil}
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
-class CheckEtmpServiceTest extends UnitSpec with MockitoSugar with BeforeAndAfterEach with GuiceOneAppPerSuite {
+class CheckEtmpServiceTest extends UnitSpec with MockitoSugar with BeforeAndAfterEach {
```
Comment (paulanderson1234) [inline test/services/CheckEtmpServiceTest.scala:36]: We can remove this from the imports if not used + other unused imports eg TestUtil

File: app/models/StatusNotificationModel.scala (hunk 6 above)
Comment (jmountain-hmrc) [reply app/models/StatusNotificationModel.scala:47]: Yes

File: test/forms/GroupMemberDetailsFormTest.scala [hunk 12]
```diff
@@ -25,7 +25,7 @@ import org.scalatest.mockito.MockitoSugar
 import org.scalatestplus.play.OneServerPerSuite
 import uk.gov.hmrc.play.test.UnitSpec
 
-class GroupMemberDetailsFormTest extends UnitSpec with MockitoSugar with OneServerPerSuite {
+class GroupMemberDetailsFormTest extends UnitSpec with MockitoSugar {
```
Comment (paulanderson1234) [inline test/forms/GroupMemberDetailsFormTest.scala:28]: Unused imports now no longer using

Can we also update MockittoSugar to use import org.scalatestplus.mockito.MockitoSugar ?

File: test/controllers/ApplicationStatusControllerTest.scala [hunk 13]
```diff
@@ -69,8 +68,9 @@ class ApplicationStatusControllerTest extends AwrsUnitTestTraits
     case false => TestUtil.testBusinessDetails()
   }
 
-  val testApplicationStatusController = new ApplicationStatusController(mockMCC, testStatusManagementService,
-    mockAuditable, mockAccountUtils, mockAuthConnector, testSave4LaterService, mockDeEnrolService, mockAppConfig)
+  lazy val mockTemplate = app.injector.instanceOf[views.html.awrs_application_status]
```
Comment (paulanderson1234) [inline test/controllers/ApplicationStatusControllerTest.scala:71]: Test runs without these being lazy val ... can we just use val ?

File: test/controllers/ApplicationControllerTest.scala [hunk 14]
```diff
@@ -30,7 +30,7 @@ import scala.concurrent.Future
 
 class ApplicationControllerTest extends AwrsUnitTestTraits with MockAuthConnector {
 
-  val testApplicationController = new ApplicationController(mockServicesConfig, mockMCC, mockAppConfig)
+  val testApplicationController = new ApplicationController(mockMCC, mockAppConfig)
```
Comment (paulanderson1234) [inline test/controllers/ApplicationControllerTest.scala:33]: There are unused imports that we can remove 
import play.api.{Configuration, Play}
import play.api.Mode.Mode

File: test/controllers/AlreadyStartingTradingControllerTest.scala [hunk 15]
```diff
@@ -24,11 +24,13 @@ import org.mockito.Mockito._
 import play.api.mvc.AnyContentAsFormUrlEncoded
 import play.api.test.FakeRequest
 import play.api.test.Helpers._
+import play.twirl.api.Html
 import services.DataCacheKeys._
 import services.ServicesUnitTestFixture
 import utils.TestUtil._
 import utils.{AwrsUnitTestTraits, TestUtil}
 import views.Configuration.NewApplicationMode
+import views.html.{awrs_already_starting_trading, awrs_business_directors}
```
Comment (paulanderson1234) [inline test/controllers/AlreadyStartingTradingControllerTest.scala:33]: Unused imports can be removed where possible

File: test/controllers/AddressLookupControllerTest.scala [hunk 16]
```diff
@@ -18,29 +18,29 @@ package controllers
...
 import scala.concurrent.Future
-import uk.gov.hmrc.http.{BadRequestException, HeaderCarrier}
 
 class AddressLookupControllerTest extends UnitSpec with MockitoSugar with BeforeAndAfterEach with MockAuthConnector {
 
   val request: FakeRequest[AnyContentAsEmpty.type] = FakeRequest()
   val postCode: String = testPostcode
   val invalidPostcode: String = "/*/*/"
-  val mockAddressLookupService: AddressLookupService = mock[AddressLookupService]
+  val mockAddressLookupService: AddressLookupConnector = mock[AddressLookupConnector]
```
Comment (paulanderson1234) [inline test/controllers/AddressLookupControllerTest.scala:43]: If we are changing the reference to Service can we look at refactoring this to mockAddressLookupConnector as previously mentioned

File: app/connectors/AddressLookupConnector.scala (hunk 2 above)
Comment (paulanderson1234) [reply app/connectors/AddressLookupConnector.scala]: Changed - resolved

File: app/controllers/ApplicationController.scala (hunk 3 above)
Comment (paulanderson1234) [reply app/controllers/ApplicationController.scala]: Changed - resolved

File: app/controllers/WithdrawalController.scala (hunk 5 above)
Comment (paulanderson1234) [reply app/controllers/WithdrawalController.scala]: Changed - resolved

File: app/models/StatusNotificationModel.scala (hunk 6 above)
Comment (paulanderson1234) [reply app/models/StatusNotificationModel.scala:47]: Intellij showing cannot resolve symbol code ?

File: test/views/BusinessTypeViewTest.scala (hunk 7 above)
Comment (paulanderson1234) [reply test/views/BusinessTypeViewTest.scala]: Changed - resolved

File: app/models/StatusNotificationModel.scala (hunk 6 above)
Comment (jmountain-hmrc) [reply app/models/StatusNotificationModel.scala:47]: It seems to work fine for me - also compiles correctly

File: test/utils/AwrsUnitTestTraits.scala (hunk 8 above)
Comment (paulanderson1234) [reply test/utils/AwrsUnitTestTraits.scala:22]: Changed - resolved

File: test/services/mocks/MockTestStatusManagementService.scala (hunk 9 above)
Comment (paulanderson1234) [reply test/services/mocks/MockTestStatusManagementService.scala]: Removed - resolved

File: test/controllers/AddressLookupControllerTest.scala (hunk 16 above)
Comment (paulanderson1234) [reply test/controllers/AddressLookupControllerTest.scala]: Changed - reolved

File: test/controllers/ApplicationStatusControllerTest.scala (hunk 13 above)
Comment (paulanderson1234) [reply test/controllers/ApplicationStatusControllerTest.scala]: Amended - resolved

**Pull request merged**